### PR TITLE
WIP: support background outline color and width for labels

### PR DESCRIPTION
**Pull request by jony89**
relates to issue #8907 

This is what I've tried. yet, could not see any border. tried `backgroundOutlineColor: Color.BLACK` with no results. I do see that the value is passed correctly to `LabelCollection`

What am I missing here?

This is still work in progress and I will add also `backgroundOutlineWidth` support once the color works.

@hpinkos

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @jony89!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/master/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by jony89** [comment]
@mramato

File: Source/Scene/Label.js [hunk 1]
```diff
@@ -169,6 +169,10 @@ function Label(options, labelCollection) {
     options.backgroundColor,
     new Color(0.165, 0.165, 0.165, 0.8)
   );
+  this._backgroundOutlineColor = defaultValue(
+    options.backgroundOutlineColor,
```
Comment (nmschulte) [inline Source/Scene/Label.js:173]:
```suggestion
    Color.clone(options.backgroundOutlineColor),
```

**Comment by nmschulte** [comment]
The reason this change has no effect is because `Billboard` does not support any `outline*` properties.

**Comment by jony89** [comment]
@nmschulte ofcourse why not? this how the outline for glyph works today. each letter in the LabelGraphic is transformed to Billboard, and it supports outline.

`Billboard.js` does have `_outlineColor` and `_outlineWidth` properties.

**Comment by nmschulte** [comment]
@jony89 The `outline*` aspects of `Billboard` are used to support `Label`'s `outline*` properties; they enable outlining text on labels, not outlining a billboard's background.

https://cesium.com/docs/cesiumjs-ref-doc/Label.html?classFilter=label#outlineColor

The comments in `Billboard.js` note this:
https://github.com/jony89/cesium/blob/master/Source/Scene/Billboard.js#L989

And the only other references to `outline*` are in `writeSDF` in `BillboardCollection.js`:
https://github.com/CesiumGS/cesium/blob/1.70.1/Source/Scene/BillboardCollection.js#L1537

**Comment by jony89** [comment]
@nmschulte can u direct me what needs to be done in order to support this? 

I am not pro with shaders though

**Comment by nmschulte** [comment]
I'm not familiar with how `Billboard` applies the background from `Label`, cannot direct you very well.
I think properly supporting an outline is complicated by translucency support of the background; you cannot simply draw another billboard with slightly larger dimensions and paint the existing background over it.

However, if the background is somehow drawn as a texture, maybe you could draw the outline via the texture applied to the background and appropriately increase the size of the background.  Otherwise, I guess you're correct to think about doing this with some sort of shader implementation.

**Comment by hpinkos** [comment]
@jony89 thanks for you interest in implementing this.  I'm sorry, but I don't have the bandwidth at the moment to assist in the lower level shader changes involved with completing this feature.  You would need to make changes to `BillboardCollection` to pack the values of the additional options to send to the shaders, then add support for rendering the outline to `BillboardCollectionVS.glsl` and `BillboardCollectionFS.glsl`.

I'm going to close this PR because it is incomplete.  Feel free to add comments to #8907 to track your progress with this.

**Pull request closed without merging**